src/widgets/PrimaryAndSecondaryPanel/components/PrimaryAndSecondaryPanelWidget.tsx

```diff
@@ -43,15 +56,15 @@ export function LayersAndLegendsWidget({ layersProps, legendProps }: PanelProps)
 
   useAutoCollapsePanel(isOpen, onPanelClose);
 
-  if (!layersProps && !legendProps) return null;
+  if (!primaryProps && !secondaryProps) return null;
 
-  const header = !layersProps ? legendProps!.header : layersProps.header;
-  const panelIcon = !layersProps ? legendProps!.panelIcon : layersProps.panelIcon;
+  const header = !primaryProps ? secondaryProps!.header : primaryProps.header;
```

Comment:
I don't get the idea, why not-null assertion is used here, if type says secondaryProps and its fields are optional.
Why not just secondaryProps?.header in all places?